### ISSUE TITLE
Font weight is not honored on iOS in RN 0.76

### ISSUE THREAD

**Issue by esbenvb** [facebook/react-native#47656]
### Description

After switching to new architecture, the text rendering does not seem to honor the `fontWeight` style as it used to do, on iOS.

With RN 0.76 and new architecture enabled, it's not choosing the font based on the weight on iOS, even though that's the intended behavior according to the docs: https://reactnative.dev/docs/text-style-props#fontweight

### Steps to reproduce

clone the project and run 
`yarn && npx pod-install && yarn ios`

Or add this code to an existing RN 0.76 project with new architecture enabled:

```jsx
<Text
  style={{
    fontFamily: 'RobotoCondensed-Bold',
    fontSize: 32,
    fontWeight: '700',
  }}>
    RobotoCondensed-Bold - weight 700
</Text>
<Text
    style={{
    fontFamily: 'RobotoCondensed-Regular',
    fontSize: 32,
    fontWeight: '700',
  }}>
    RobotoCondensed-Regular - weight 700
</Text>
```

### React Native Version

0.76.2

### Affected Platforms

Runtime - iOS

### Areas

Fabric - The New Renderer

### Output of `npx react-native info`

```text
System:
  OS: macOS 15.0.1
  CPU: (10) arm64 Apple M1 Max
  Memory: 819.03 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.18.2
    path: ~/Library/Caches/fnm_multishells/40214_1731916352195/bin/node
  Yarn:
    version: 1.22.22
    path: ~/Library/Caches/fnm_multishells/40214_1731916352195/bin/yarn
  npm:
    version: 9.8.1
    path: ~/Library/Caches/fnm_multishells/40214_1731916352195/bin/npm
  Watchman:
    version: 2024.10.07.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK:
    API Levels:
      - "27"
      - "28"
      - "29"
      - "31"
      - "32"
      - "33"
      - "34"
      - "34"
      - "35"
    Build Tools:
      - 28.0.3
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 34.0.0
      - 35.0.0
    System Images:
      - android-28 | Google ARM64-V8a Play ARM 64 v8a
      - android-29 | Intel x86 Atom_64
      - android-29 | Google APIs Intel x86 Atom
      - android-33 | Google APIs ARM 64 v8a
      - android-34 | Google Play ARM 64 v8a
    Android NDK: 25.0.8775105
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11255304
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 1.8.0_292
    path: /usr/bin/javac
  Ruby:
    version: 3.3.5
    path: /opt/homebrew/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.1
    wanted: 15.0.1
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.2
    wanted: 0.76.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
N/A
```

### Reproducer

https://github.com/esbenvb/rn-font-weight-issue

### Screenshots and Videos

✅ Without new architecture - working as expected
![Simulator Screenshot - iPhone 15 - 2024-11-18 at 09 00 58](https://github.com/user-attachments/assets/9c03a81c-1833-4ad2-8472-c7fe47bf4943)

❌ With new architecture - not working as expected 
![Simulator Screenshot - iPhone 15 - 2024-11-18 at 09 03 57](https://github.com/user-attachments/assets/c5dad95d-2e21-44a6-8dfe-148ecb28d11d)

**Comment by zhongwuzw** [comment]
Hi, it may fixed by #47691.

**Comment by orkidaW** [comment]
I am facing font issue in react native 0.76 too. The google custom font added doesn't get recognized in IOS ,but works fine in android.

**Comment by esbenvb** [comment]
@zhongwuzw 

The patch in the PR seems to work for me. What's the timeline for releasing this in 0.76.3 ?

<img width="565" alt="image" src="https://github.com/user-attachments/assets/00c8a627-29bf-4bf1-9a89-4f8b414badfb">

**Comment by esbenvb** [comment]
I'm afraid I was too quick here... When switching to our main app that uses the Chivo font, it seems like it does not show the Chivo-Regular properly... I have updated the repro project.

With your patch and new architecture
<img width="565" alt="image" src="https://github.com/user-attachments/assets/0f3a4074-c5bf-4946-ad37-4ad128026ca5">

Without new architecture
<img width="565" alt="image" src="https://github.com/user-attachments/assets/70d3febc-8486-4092-8af4-983eea0ae1ed">

**Comment by cipolleschi** [comment]
The problem with Chivo's font is that metadata are a bit off. The weight metadata for each font returns the same weight, therefore the current algorithm can't find the right one.

The issue can be fixed if we don't try to match the weight but we match the name exactly if it exists. I can provide a fix for it tomorrow.

**Comment by esbenvb** [comment]
That's strange!

But nevertheless, this font seems to work for Android and also for iOS with the old architecture?

**Comment by zhongwuzw** [comment]
@esbenvb May fixed in #47742

**Comment by esbenvb** [comment]
Seems to work!
